packages/next/server/dev/on-demand-entry-handler.ts

```diff
@@ -250,29 +198,24 @@ export default function onDemandEntryHandler(
         })
       }
 
-      const isClientOrMiddleware = clientOnly || isMiddleware
-
-      const promise = isApiRoute
-        ? addPageEntry('server')
-        : isClientOrMiddleware
-        ? addPageEntry('client')
-        : Promise.all([
-            addPageEntry('client'),
-            addPageEntry(
-              isEdgeServer && !isCustomError ? 'edge-server' : 'server'
-            ),
-          ])
+      const promises = runDependingOnPageType({
+        page,
+        pageRuntime: await getPageRuntime(absolutePagePath, nextConfig),
+        onClient: () => addPageEntry('client'),
+        onServer: () => addPageEntry('server'),
+        onEdgeServer: () => addPageEntry('edge-server'),
+      })
 
       if (entriesChanged) {
         reportTrigger(
-          isApiRoute || isMiddleware || clientOnly
-            ? normalizedPage
-            : `${normalizedPage} (client and server)`
+          clientOnly || promises.length > 1
```

Comment:
If `clientOnly` is true here should `(client and server)` be shown in the compilation message?

Reply:
Nice catch, I think this is wrong. `clientOnly` should only show the page and `promises.length > 1` should indeed should both so the condition has to be `!clientOnly && promises.length > 1`. Updating it!